lib/datadog/security/reactive/engine.rb

```diff
@@ -0,0 +1,41 @@
+require 'datadog/security/reactive/address_hash'
+require 'datadog/security/reactive/subscriber'
+
+module Datadog
+  module Security
+    module Reactive
+      class Engine
+        def initialize
+          @data = {}
+          @subscribers = AddressHash.new { |h, k| h[k] = [] }
```

Comment:
Minor: Since `AddressHash` always seems to be used with this new array auto-initialization, perhaps this could be moved to the class itself?